### PR TITLE
Appsody Operator must-gather

### LINKED ISSUE #148: Add must gather scripts for Appsody
Add must-gather scripts to Appsody to help troubleshoot issues. As part of this, [doc/troubleshooting.md](https://github.com/appsody/appsody-operator/blob/master/doc/troubleshooting.md) should be updated to help users find problems using the must-gather scripts.

### PR DESCRIPTION
**Pull request by JusteenR**
**What this PR does / why we need it?**:

- `Appsody Operator must-gather` is a tool built on top of OpenShift must-gather that expands its capabilities to gather information about the Appsody Operator.
- Creates a local directory with a dump of relevant troubleshooting information
- Includes information about the pods that are not in the ready state, appsody operator pod(s), knative, appsody application, and the resources that are managed by the operator

**Does this PR introduce a user-facing change?**

- [x] User guide
- [x] `CHANGELOG.md`

**Which issue(s) this PR fixes**:

Fixes #148

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -44,3 +46,14 @@ jobs:
     - docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
     - make push-image
     if: branch = master AND fork = false AND type != pull_request AND env(DOCKER_USERNAME) IS NOT blank AND env(DOCKER_PASSWORD) IS NOT blank
+  - name: Build must gather image
+    stage: build must gather
+    script:
+      - make build-must-gather
+  - name: Build must gather image and push
```
Comment (navidsh) [inline .travis.yml]: Since this stage is doing both build and push, do we still need `Build must gather image` stage?

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,10 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Added
+
+- Added must-gather scripts for troubleshooting. ([#148](https://github.com/appsody/appsody-operator/issues/148))
```
Comment (navidsh) [inline CHANGELOG.md]: This should point to PR(s). Please update the link.

File: doc/troubleshooting.md [hunk 3]
```diff
@@ -62,4 +62,20 @@ If the operator is running fine, check the status of the `AppsodyApplication` Cu
...
+
+`Appsody Operator must-gather` is a tool built on top of OpenShift must-gather that expands its capabilities to gather information about the Appsody Operator.
+
+### Usage
+
+```console
+$ oc adm must-gather --image=docker.io/appsody/application-operator:daily-must-gather
+```
+
+Note: `must-gather` flag is a new feature added to oc v4.x. If you are using an older version of oc, you can get a new version of the CLI from here. You can use oc v4.x against 3.11 cluster.
```
Comment (navidsh) [inline doc/troubleshooting.md]: Please use "OpenShift client CLI" instead of "oc".

Comment (navidsh) [reply doc/troubleshooting.md]: The link in "...CLI from here..." is missing.

File: must-gather/README.md [hunk 4]
```diff
@@ -0,0 +1 @@
+# Appsody Operator must-gather
```
Comment (navidsh) [inline must-gather/README.md]: You can add text to refer people to the troubleshooting doc

File: CHANGELOG.md (hunk 2 above)
Comment (navidsh) [reply CHANGELOG.md]: Also seems like there is a conflict with the master branch in this file

File: .travis.yml (hunk 1 above)
Comment (JusteenR) [reply .travis.yml]: I was initially following the the operator build step but it isn't necessary for this so I removed the `Build must gather image` stage.

File: CHANGELOG.md (hunk 2 above)
Comment (JusteenR) [reply CHANGELOG.md]: Updated the link and fixed the conflicts.

File: doc/troubleshooting.md (hunk 3 above)
Comment (JusteenR) [reply doc/troubleshooting.md]: Changed oc to OpenShift client CLI and added the link.

**Pull request merged**